src/components/include/protocol_handler/session_observer.h

```diff
@@ -54,6 +54,58 @@ namespace protocol_handler {
  */
 enum { HASH_ID_NOT_SUPPORTED = 0, HASH_ID_WRONG = 0xFFFF0000 };
 
+/**
+ * @brief Struct with data containing attributes of starting session
+ **/
+struct StartingSessionContext {
+  transport_manager::ConnectionUID connection_id_;
+  uint8_t initial_session_id_;
+  uint32_t new_session_id_;
+  protocol_handler::ServiceType service_type_;
+  uint32_t hash_id_;
+  bool is_protected_;
+  bool is_new_service_;
+  bool is_ptu_required_;
+
+  /**
+   * @brief Constructor
+   */
+  StartingSessionContext()
+      : connection_id_(0)
+      , initial_session_id_(0)
+      , new_session_id_(0)
+      , service_type_(protocol_handler::kInvalidServiceType)
+      , hash_id_(0)
+      , is_protected_(false)
+      , is_new_service_(false)
+      , is_ptu_required_(false) {}
+
+  /**
+   * @brief Constructor
+   * @param connection_id_ Connection identifier within which session is
+   * started.
+   * @param session_id Session ID specified to OnSessionStartedCallback()
+   * @param new_session_id Session ID generated
+   * @param service_type Type of service
+   * @param hash_id Hash ID generated from connection_handle and
+   * new_session_id
+   * @param is_protected Whether service will be protected
+   * @param is_new_service Whether service was already established
+   **/
+  StartingSessionContext(transport_manager::ConnectionUID connection_id,
```

Comment:
using this constructor, what's the value of is_new_service_ and is_ptu_required_?

Reply:
@yang1070 is_new_service_ and is_ptu_required_ have type bool. C++ initialize all non-class types with zeroes (so-called zero initialization) in case if they was not listed in initializer list. For boolean it is a false value. However if you insist on that, I could list these values in initializer list explicitly.

Reply:
@yang1070 I made some updates in StartingSessionContext constructors. Now is_new_service_ and is_ptu_required_  are specified in initialization list explicitly. Please see commit 36306024c326aad930a636931f64f00928fce51e